### DISCUSSION TITLE
Including deleteMany and createMany nested writes in one transaction (based on a list of IDs)

### DISCUSSION THREAD

**Discussion by hyunwoona** [prisma/prisma#15514] [Q&A] [answered]
Hi guys!
In my model, I have `User`, and `SkinConcern`.

```
model User {
  id                 String                 @id @default(cuid())
  nickname           String?                @unique
  skinConcerns       SkinConcernsForUsers[]
}

model SkinConcern {
  id   String                 @id
  user SkinConcernsForUsers[]
}
```
Since it's a many-to-many relation (User A's concern can be pigmentation and acne, and User B's concern can be pigmentation and wrinkles. Users can have multiple skin concerns, and skin concerns can be shared by multiple users) I created a relation table.

```
model SkinConcernsForUsers {
  user          User        @relation(fields: [userId], references: [id])
  userId        String
  skinConcern   SkinConcern @relation(fields: [skinConcernId], references: [id])
  skinConcernId String

  @@id([userId, skinConcernId])
}
```

In the UI, I have checkboxes to let users select/deselect their skin concerns.

I'm trying to update skinConcerns in one `prisma.user.update` run.

Right now I'm running two extra commands to update them.
Not only is it extra, but it's also two extra commands: First, delete all existing entries that user selected/deselected in the UI. Second, add all entries with `skinDuplicates: true`.
In the code below, `skinConcerns` contains the set of `SkinConcern` IDs that the user selected/deselected.

```
      const skinConcernsDeleteResult =
        await prisma.skinConcernsForUsers.deleteMany({
          where: {
            userId,
            NOT: {
              skinConcernId: {
                in: skinConcerns,
              },
            },
          },
        });

      const skinConcernCreateResult =
        await prisma.skinConcernsForUsers.createMany({
          data: skinConcerns.map((skinConcern) => ({
            userId,
            skinConcernId: skinConcern,
          })),
          skipDuplicates: true,
        });
```

Is there a way to achieve this in one command? Or ideally, is there a way to do this in the same `prisma.user.update` command?
I appreciate any help. Thank you!

**Comment by hyunwoona** [comment, marked as answer]
I solved this by passing both `deleteMany` and `createMany` in the same transaction as nested writes. I wasn't sure if this was a valid thing to do, but it works as intended.

```
      const userUpdateResult = await prisma.user.update({
        where: { id: userId },
        data: {
          // ...
          skinConcerns: {
            deleteMany: {
              userId,
              NOT: {
                skinConcernId: {
                  in: skinConcerns,
                },
              },
            },
            createMany: {
              data: skinConcerns.map((skinConcern) => ({
                skinConcernId: skinConcern,
              })),
              skipDuplicates: true,
            },
          },
        },
        // ...
      });
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
